### PR TITLE
Updates store json rpc api

### PR DESCRIPTION
**Pull request by staheri14**
Closes https://github.com/status-im/nim-waku/issues/489
The corresponding specs PR https://github.com/vacp2p/rfc/pull/356

### CODE REVIEW

File: tests/v2/test_jsonrpc_waku.nim [hunk 1]
```diff
@@ -259,7 +259,7 @@ procSuite "Waku v2 JSON-RPC API":
     let client = newRpcHttpClient()
     await client.connect("127.0.0.1", rpcPort)
 
-    let response = await client.get_waku_v2_store_v1_messages(@[defaultContentTopic], some(StorePagingOptions()))
+    let response = await client.get_waku_v2_store_v1_messages(@[HistoryContentFilter(contentTopic: defaultContentTopic)], some(StorePagingOptions()))
```
Comment (oskarth) [inline tests/v2/test_jsonrpc_waku.nim:262]: What happens if we don't pass any content filter here?

**Review by jm-clius — APPROVED** [review]
LGTM!

File: tests/v2/test_jsonrpc_waku.nim (hunk 1 above)
Comment (jm-clius) [reply tests/v2/test_jsonrpc_waku.nim:262]: I think the underlying `store` still expects a `HistoryQuery` with populated contentFilters, but I also think changing to an `Option` will be an improvement.

Comment (staheri14) [reply tests/v2/test_jsonrpc_waku.nim:262]: The store json rpc api was designed this way with `topics` filed which is now `contentFilters`  being mandatory. I also agree that making it optional makes sense and would also match the store protocol api.

Comment (staheri14) [reply tests/v2/test_jsonrpc_waku.nim:262]: Opened an issue re making it optional: https://github.com/status-im/nim-waku/issues/505

**Review by oskarth — APPROVED** [review]

**Pull request merged**